File: physio-server/PhysioBoo.Domain/Enums/Role.cs
using PhysioBoo.SharedKernel.Attributes;

namespace PhysioBoo.Domain.Enums
{
    public enum Role
    {
        [RoleMetadata("Super Administrator", "Full system access", true)]
        SUPER_ADMIN,

        [RoleMetadata("Administrator", "System management & configuration")]
        ADMIN,

        [RoleMetadata("Nurse", "Nursing and patient care operations")]
        NURSE,

        [RoleMetadata("Pharmacist", "Drug management & dispensing")]
        PHAMACIST,

        [RoleMetadata("Cashier", "Billing & payment handling")]
        CASHIER,

        [RoleMetadata("Lab Technician", "Lab tests & results processing")]
        LAB_TECHNICIAN,

        [RoleMetadata("Radiologist", "Radiology reporting")]
        RADIOLOGIST,

        [RoleMetadata("Receptionist", "Front desk & appointment handling")]
        RECEPTIONIST,

        [RoleMetadata("Accountant", "Finance and invoice management")]
        ACCOUNTANT,

        [RoleMetadata("Inventory Manager", "Warehouse & inventory operations")]
        INVENTORY_MANAGER,

        [RoleMetadata("IT Support", "Technical system support")]
        IT_SUPPORT,

        [RoleMetadata("Doctor", "Medical diagnosis and treatment", isPublicForRegistration: true)]
        DOCTOR,

        [RoleMetadata("Patient", "Health service users and personal records tracking.", isPublicForRegistration: true)]
        PATIENT
    }
}

File: physio-mobile/src/app/shared/enums/role.ts
/** Ported verbatim from physio-app/src/app/shared/enums/role.ts — must stay in sync with PhysioBoo.Domain.Enums.Role. */
export enum Role {
    SUPER_ADMIN = "SUPER_ADMIN",
    ADMIN = "ADMIN",
    NURSE = "NURSE",
    PHAMACIST = "PHAMACIST",
    CASHIER = "CASHIER",
    LAB_TECHNICIAN = "LAB_TECHNICIAN",
    RADIOLOGIST = "RADIOLOGIST",
    RECEPTIONIST = "RECEPTIONIST",
    ACCOUNTANT = "ACCOUNTANT",
    INVENTORY_MANAGER = "INVENTORY_MANAGER",
    IT_SUPPORT = "IT_SUPPORT",
    DOCTOR = "DOCTOR",
    PATIENT = "PATIENT"
}

/** Roles this mobile sub-project (Foundation + Guest + Patient) serves. Staff/manager/admin roles route to a "use the desktop app" notice for now. */
export const MOBILE_SUPPORTED_ROLES: readonly Role[] = [Role.PATIENT];
